### PR TITLE
#1036: Added pytorch and numba integration test compatible with CPU only VM

### PR DESCRIPTION
**Pull request by tomuben**
related https://github.com/exasol/script-languages-release/issues/1036

### CODE REVIEW

File: test_container/tests/test/python3-cuda-flavor/pytorch.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                    model.eval()
+                    with torch.no_grad():
+                        y_pred = model(x_train)
+                        mse = criterion(y_pred, y_train)
+                        return f'Mean Squared Error: {mse.item():.4f}'
+                /
+                '''))
+
+        row = self.query(f"SELECT pytorchbasic.test_pytorch(1000);")[0]
+        self.assertIn('Mean Squared Error', row[0])
```
Comment (tkilias) [inline test_container/tests/test/python3-cuda-flavor/pytorch.py:68]: I would assert if the loss is not NaN, this could hint to something is broken, if it worked before

File: test_container/tests/test/python3-cuda-flavor/numba.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+                    B = np.array(np.random.sample(N) + 10, dtype=dtype)
+                    C = np.array(np.random.sample(N), dtype=dtype)
+                
+                    D = cu_discriminant(A, B, C)
+                
+                    return str(D)
+                /
+                '''))
+
+        row = self.query("SELECT numbabasic.test_numba(10000);")[0]
```
Comment (kaklakariada) [inline test_container/tests/test/python3-cuda-flavor/numba.py:43]: Is it necessary to verify the returned value?

**Review by kaklakariada — CHANGES_REQUESTED** [review]

File: test_container/tests/test/python3-cuda-flavor/numba.py (hunk 2 above)
Comment (tomuben) [reply test_container/tests/test/python3-cuda-flavor/numba.py:43]: I think not. The return value is some random number.  If there is any problem, an exception is thrown and this breaks the test.

File: test_container/tests/test/python3-cuda-flavor/pytorch.py [hunk 3]
```diff
@@ -60,12 +61,13 @@ def forward(self, x):
...
                         mse = criterion(y_pred, y_train)
-                        return f'Mean Squared Error: {mse.item():.4f}'
+                        return mse.item()
                 /
                 '''))
 
         row = self.query(f"SELECT pytorchbasic.test_pytorch(1000);")[0]
-        self.assertIn('Mean Squared Error', row[0])
+
+        self.assertFalse(math.isnan(row[0]))
```
Comment (tkilias) [inline test_container/tests/test/python3-cuda-flavor/pytorch.py:70]: I think, NaN is transformed to NULL in the DB

File: test_container/tests/test/python3-cuda-flavor/numba.py (hunk 2 above)
Comment (tkilias) [reply test_container/tests/test/python3-cuda-flavor/numba.py:43]: In that case your are assert should be , did we get a row back

File: test_container/tests/test/python3-cuda-flavor/pytorch.py (hunk 3 above)
Comment (tomuben) [reply test_container/tests/test/python3-cuda-flavor/pytorch.py:70]: Appearantly the build failed when fetching micromamba. Running the test locally , the assert is correct.

File: test_container/tests/test/python3-cuda-flavor/numba.py (hunk 2 above)
Comment (tomuben) [reply test_container/tests/test/python3-cuda-flavor/numba.py:44]: Added an assert for the resulting row

**Review by kaklakariada — APPROVED** [review]

**Review by tkilias — APPROVED** [review]
